Doc/reference/expressions.rst

```diff
@@ -28,13 +28,13 @@ Arithmetic conversions
 .. index:: pair: arithmetic; conversion
 
 When a description of an arithmetic operator below uses the phrase "the numeric
-arguments are converted to a common type", this means that the operator
+arguments are converted to a common real type", this means that the operator
 implementation for built-in types works as follows:
 
-* If either argument is a complex number, the other is converted to complex;
+* If both arguments are complex numbers, no conversion is performed;
 
-* otherwise, if either argument is a floating-point number, the other is
-  converted to floating point;
+* otherwise, if either argument is a complex or a floating-point number, the
+  other is converted to a floating-point number;
```

Comment:
```suggestion
* if either argument is a complex or a floating-point number, the other is converted to a floating-point number;
```

I think it's better to have "if/if/otherwise" rather than "if/otherwise/otherwise". AFAIU, this means that (up to permutations):

- (complex, complex) -> (complex, complex)
- (complex, int/float) -> (complex, float) [**this is the new case**]
- (float, int/float) -> (float, float)
- (int, int) -> (int, int)

I think this behaviour is well explained and this is what we expect if we want mixed-arithmetic.